commands/hugo_test.go

```diff
@@ -0,0 +1,305 @@
+	"testing"
+
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"github.com/spf13/hugo/helpers"
+	"github.com/spf13/hugo/hugofs"
+	"github.com/spf13/hugo/hugolib"
+	"github.com/spf13/viper"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestUserErrorCommandNewUserError(t *testing.T) {
+	expectedUserError := commandError{"test\n", true}
+	actualUserError := newUserError("test")
+	assert.Equal(t, expectedUserError, actualUserError)
+}
+
+func TestSystemErrorCommandNewSystemError(t *testing.T) {
+	expectedSystemError := commandError{"test\n", false}
+	actualSystemError := newSystemError("test")
+	assert.Equal(t, expectedSystemError, actualSystemError)
+}
+
+func TestUserErrorCommandNewFormattedUserError(t *testing.T) {
+	expectedUserError := commandError{"test:12", true}
+	actualUserError := newUserErrorF("test:%d", 12)
+	assert.Equal(t, expectedUserError, actualUserError)
+}
+
+func TestSystemErrorCommandNewFormattedSystemError(t *testing.T) {
+	expectedSystemError := commandError{"test:12", false}
+	actualSystemError := newSystemErrorF("test:%d", 12)
+	assert.Equal(t, expectedSystemError, actualSystemError)
+	if actualSystemError != expectedSystemError {
+		t.Errorf("Actual (%v) returned command did not equal expected (%v) command.", actualSystemError, expectedSystemError)
+	}
+}
+
+func TestUserErrorReturnError(t *testing.T) {
+	cError := commandError{"test error", false}
+	expectedError := "test error"
+	actualError := cError.Error()
+	assert.Equal(t, expectedError, actualError)
+}
+
+func TestIsUserError(t *testing.T) {
+	var errors = []struct {
+		err error
+	}{
+		{fmt.Errorf("argument")},
+		{fmt.Errorf("flag")},
+		{fmt.Errorf("shorthand")},
+		{commandError{"test", true}},
+	}
+
+	for _, v := range errors {
+		if !isUserError(v.err) {
+			t.Errorf("Error (%v) should have been flagged as a user error.", v)
+		}
+	}
+}
+
+func TestExecuteAddedCommands(t *testing.T) {
+	var defaultCommands = []struct {
+		cmd *cobra.Command
+	}{
+		{serverCmd},
+		{versionCmd},
+		{configCmd},
+		{checkCmd},
+		{benchmarkCmd},
+		{convertCmd},
+		{newCmd},
+		{listCmd},
+		{undraftCmd},
+		{importCmd},
+		{genCmd},
+	}
+
+	basepath := filepath.Join(os.TempDir(), "blog")
+	configPath := filepath.Join(basepath, "config.toml")
+	hugofs.SourceFs = new(afero.OsFs)
+	hugofs.SourceFs.MkdirAll(basepath, os.ModePerm|os.ModeDir)
+	paths := []string{
+		filepath.Join(basepath, "public"),
+		filepath.Join(basepath, "layouts"),
+		filepath.Join(basepath, "content"),
+		filepath.Join(basepath, "archetypes"),
+		filepath.Join(basepath, "static"),
+		filepath.Join(basepath, "data"),
+	}
+
+	for _, path := range paths {
+		hugofs.SourceFs.MkdirAll(path, os.ModePerm|os.ModeDir)
+	}
+
+	file, err := hugofs.SourceFs.Create(configPath)
+	if err != nil {
+		t.Error(err.Error())
+	}
+	err = hugofs.SourceFs.Chmod(configPath, os.ModeTemporary|os.ModePerm)
+	if err != nil {
+		t.Error(err.Error())
+	}
+	basicConfig := `
+        contentdir = "` + string(paths[2]) + `"
```

Comment:
I need to escape path here so that it doesn't cause any problem on a non-unix system.